lib/internal/source_map/prepare_stack_trace.js

```diff
@@ -81,11 +81,12 @@ const prepareStackTrace = (globalThis, error, trace) => {
             );
           }
           // Show both original and transpiled stack trace information:
+          const prefix = name ? `\n        -> at ${name}` : '\n        ->';
           const originalSourceNoScheme =
             StringPrototypeStartsWith(originalSource, 'file://') ?
               fileURLToPath(originalSource) : originalSource;
-          str += `\n        -> ${originalSourceNoScheme}:${originalLine + 1}:` +
-            `${originalColumn + 1}`;
+          str += `${prefix} (${originalSourceNoScheme}:${originalLine + 1}:` +
```

Comment:
Worth noting that this puts a `(`, `)` around the file paths in supplemental stack traces now. I felt that this looked better when we have an alternate symbol name, and it seemed more consistent to always wrap in parenthesis:


```js
Error: an exception
    at branch (/Users/bencoe/oss/node-1/test/fixtures/source-map/typescript-throw.js:20:15)
        -> (/Users/bencoe/oss/node-1/test/fixtures/source-map/typescript-throw.ts:18:11)
```

Reply:
as a heads up, this might make it incompatible with the tc39 stacks proposal, if it ever lands - it’ll only specify the union of what browsers do.

Reply:
@ljharb what would you suggest we do to provide appropriate context about the transpiled call site and original call site, in the spirit of the spirit of the tc39 stacks proposal?

I'm hoping to make the case for removing the experimental language from `--enable-source-maps` soon -- so now would be a good time to make significant changes to output if needed.

Reply:
The challenge is specifying what browsers all already do, without permitting anything beyond that.

What do browsers do here on source mapped exception stacks? If the answer is “nothing special”, then that’s probably what node should do too :-/

Reply:
> What do browsers do here on source mapped exception stacks? If the answer is “nothing special”, then that’s probably what node should do too :-/

TypeScript and other transpiled flavors or JavaScript have exploded in popularity over the past several years, I'd argue that stack traces just haven't caught up with this need:

* [6.1m](https://github.com/evanw/node-source-map-support/network/dependents?package_id=UGFja2FnZS00OTI3ODk0NTI%3D) projects rely `source-map-support`, to provide better debugging information in stack traces are used. 
* A [healthy number of folks](https://github.com/search?p=1&q=%22--enable-source-maps%22&type=Code) are finding `--enable-source-maps` (_my team is [using it ourselves](https://github.com/googleapis/nodejs-storage/blob/master/.mocharc.js#L15) because it speeds up our debugging during test).
* Threads [like this](https://twitter.com/brianleroux/status/1326886967020191747?s=20) indicate that folks are more concerned about transpiled code, when it comes to using contextual information about the stack.

I would have originally gone with the approach `source-map-support`, of simply replacing the transpiled Call Site with the original Call Site. I believe @hashseed had the idea of displaying both the source and original Call Site, and I believe there's value in this, for a variety of reasons: when looking at an error, it can be good to see how the source was transpiled; source maps, despite best efforts, are sometimes off.

---

It would be nice if the stacks proposal gave us enough flexibility that we could insert this extra contextual information into the stack trace, without breaking upstream parser.

Couldn't we just consider this to be one line of the stack trace:

```
at o (/Users/bencoe/oss/node-sourcemap-repro/dist/main.js:1:970)
      -> at functionD (webpack:///index.js:14:9)
```

☝️ the tabbed in line without an `at ` is fairly distinct.

Reply:
@ljharb let's move this conversation to the TC39 proposal; this PR is just patching a bug, not introducing new behavior to Node.js.

Reply:
per our offline discussion, it might be simpler for now to leave `.stack` as having only one of the results (source, or transpiled), and maybe providing a separate API (`Error.getOriginalStack()`, `Error.getUnmappedStack()`) to access the other one? That would ensure no conflicts with the stacks proposal.

Reply:
Did this land as-is, with the parentheticals discussed here?